src/utils.c

```diff
@@ -115,25 +140,322 @@ SEXP colnamesInt(SEXP x, SEXP cols, SEXP check_dups) {
+    SEXP ricols = PROTECT(eval(LCONS(sym_Reduce, LCONS(sym_intersect, LCONS(lpat, R_NilValue))), rho)); protecti++;
+    LOGICAL(with)[0] = 0;
+    UNPROTECT(protecti);
+    return colnamesInt(x, ricols, ScalarLogical(false), ScalarLogical(inverse)); // handle inverse
+  }
+  // dotdot prefix handling
+  if (mode_j && (isSymbol(expr) || isLanguage(expr)) && (!peeled || inverse)) { // `|| inverse` to handle !(..cols)
+    // all(substring(av,1L,2L)=="..")
+    SEXP jexpr = PROTECT(eval(lang2(install("expression"), expr), R_GlobalEnv)); protecti++;
+    SEXP av = PROTECT(eval(lang2(install("all.vars"), jexpr), R_GlobalEnv)); protecti++;
+    if (length(av)) {
+      SEXP av12 = PROTECT(eval(lang4(install("substring"), av, ScalarInteger(1), ScalarInteger(2)), R_GlobalEnv)); protecti++;
+      SEXP avdd = PROTECT(eval(lang3(install("=="), av12, ScalarString(PRINTNAME(install("..")))), R_GlobalEnv)); protecti++;
+      SEXP alldd = PROTECT(eval(lang2(install("all"), avdd), R_GlobalEnv)); protecti++;
+      if (LOGICAL(alldd)[0]) {
+        SEXP ddname=R_NilValue, name=R_NilValue, name_exists=R_NilValue, ddname_exists=R_NilValue;
+        SEXP av3 = PROTECT(eval(lang3(install("substring"), av, ScalarInteger(3)), R_GlobalEnv)); protecti++;
+        for (int i=0; i<length(av); i++) {
+          ddname = STRING_ELT(av, i);
+          name = STRING_ELT(av3, i); // PROTECT(eval(lang3(install("substring"), ddname, ScalarInteger(3)), R_GlobalEnv)); protecti++;
+          if (!strcmp(CHAR(name), ""))
+            error("The symbol .. is invalid. The .. prefix must be followed by at least one character.");
+          name_exists = PROTECT(eval(lang3(install("exists"), ScalarString(name), rho), R_GlobalEnv)); protecti++;
+          ddname_exists = PROTECT(eval(lang3(install("exists"), ScalarString(ddname), rho), R_GlobalEnv)); protecti++;
+          if (!LOGICAL(name_exists)[0]) {
+            if (!LOGICAL(ddname_exists)[0])
+              error("Variable '%s' is not found in calling scope. Looking in calling scope because you used the .. prefix.", CHAR(name));
+            else
+              error("Variable '%s' is not found in calling scope. Looking in calling scope because you used the .. prefix. Variable '..%S' does exist in calling scope though, so please just removed the .. prefix from that variable name in calling scope.", CHAR(STRING_ELT(name, 0)), CHAR(STRING_ELT(name, 0)));
+          } else if (LOGICAL(ddname_exists)[0])
+            warning("Both '%s' and '..%s' exist in calling scope. Please remove the '..%s' variable in calling scope for clarity.", CHAR(name), CHAR(name), CHAR(name));
+        }
+        SEXP lapply = lang4(install("lapply"), lang3(install("setNames"), av3, av), install("get"), /*pos=*/rho);
+        SET_TAG(CDDDR(lapply), install("pos"));
+        SEXP env = eval(lapply, rho);
+        SEXP evaleval = lang3(install("substitute"), expr, env);
+        SEXP cols = eval(eval(evaleval, rho), rho); // huh...
```

Comment:
> SEXP cols = eval(eval(evaleval, rho), rho); // huh...

+1.

So, I understand we need to replace ```c(..cols1, ..cols2)```, did you get as far as ```c(..cols1, lapply(.SD, sum))```? I see a little farther you reference the clash between the two modes.

Reply:
I don't think so it did worked like this already